packages/web/server/api/edge-routers/one-click-trading-router.ts

```diff
@@ -40,6 +44,49 @@ export const oneClickTradingRouter = createTRPCRouter({
   getNetworkFeeLimitStep: publicProcedure.query(async () =>
     getNetworkFeeLimitStep()
   ),
+  getSessionAuthenticator: publicProcedure
+    .input(z.object({ userOsmoAddress: z.string(), publicKey: z.string() }))
+    .query(async ({ input }) => {
```

Comment:
Could the logic in this query be pulled into a `getSessionAuthenticator` funciton in complex folder? May make it more reusable, especially if we don't throw tRPC-specific errors.